arduino-ide-extension/src/browser/contributions/sketchbook.ts

```diff
@@ -1,44 +1,55 @@
-import { injectable } from '@theia/core/shared/inversify';
 import { CommandHandler } from '@theia/core/lib/common/command';
-import { MenuModelRegistry } from './contribution';
+import { Disposable } from '@theia/core/lib/common/disposable';
+import { nls } from '@theia/core/lib/common/nls';
+import { injectable } from '@theia/core/shared/inversify';
+import { SketchesError } from '../../common/protocol';
 import { ArduinoMenus } from '../menu/arduino-menus';
 import { Examples } from './examples';
-import { SketchContainer, SketchesError } from '../../common/protocol';
 import { OpenSketch } from './open-sketch';
-import { nls } from '@theia/core/lib/common/nls';
 
 @injectable()
 export class Sketchbook extends Examples {
   override onStart(): void {
     this.sketchServiceClient.onSketchbookDidChange(() => this.update());
+    this.configService.onDidChangeSketchDirUri(() => this.update());
   }
 
   override async onReady(): Promise<void> {
     this.update();
   }
 
   protected override update(): void {
-    this.sketchService.getSketches({}).then((container) => {
-      this.register(container);
-      this.menuManager.update();
-    });
-  }
-
-  override registerMenus(registry: MenuModelRegistry): void {
-    registry.registerSubmenu(
+    this.toDispose.dispose();
+    this.menuRegistry.registerSubmenu(
       ArduinoMenus.FILE__SKETCHBOOK_SUBMENU,
       nls.localize('arduino/sketch/sketchbook', 'Sketchbook'),
       { order: '3' }
     );
-  }
-
-  private register(container: SketchContainer): void {
-    this.toDispose.dispose();
-    this.registerRecursively(
-      [...container.children, ...container.sketches],
-      ArduinoMenus.FILE__SKETCHBOOK_SUBMENU,
-      this.toDispose
+    this.toDispose.push(
+      Disposable.create(() =>
+        this.menuRegistry.unregisterMenuNode(
+          ArduinoMenus.FILE__SKETCHBOOK_SUBMENU[
+            ArduinoMenus.FILE__SKETCHBOOK_SUBMENU.length - 1
+          ]
```

Comment:
TODO: code-comment that this assumes how Theia calculates the submenu ID from the menu path.

Reply:
https://github.com/arduino/arduino-ide/pull/1774/commits/40bdae69e5f9c0f814a41565aac9fc804b91dafa